core/src/main/java/de/danoeh/antennapod/core/cast/SwitchableMediaRouteActionProvider.java

```diff
@@ -0,0 +1,36 @@
+package de.danoeh.antennapod.core.cast;
+
+import android.content.Context;
+import android.support.v7.app.MediaRouteActionProvider;
+
+/**
+ * <p>Action Provider that extends {@link MediaRouteActionProvider} and allows the client to
+ * disable completely the button by calling {@link #setEnabled(boolean)}.</p>
+ *
+ * <p>It is disabled by default, so if a client wants to initially have it enabled it must call
+ * <code>setEnabled(true)</code>.</p>
+ */
+public class SwitchableMediaRouteActionProvider extends MediaRouteActionProvider {
```

Comment:
This is what allows us to easily control the cast button visibility based on the user preferences. It is necessary because it internally overrides its visibility based on available connections, so we'd have no other way of controlling it externally other than killing it completely.